test/test_pullparser.rb

```diff
@@ -82,6 +82,28 @@ def test_character_references
       end
     end
 
+    def test_text_content_with_line_breaks
+      source = "<a>A</a><b>B\n</b><c>C\r\n</c>"
```

Comment:
Ah, this is an invalid XML. We should not have multiple root elements.

Reply:
(We should report an error for this case.)

Reply:
Oh....